src/core/components/dag.js

```diff
@@ -9,6 +9,21 @@ const flattenDeep = require('lodash.flattendeep')
 module.exports = function dag (self) {
   return {
     put: promisify((dagNode, options, callback) => {
+      if (typeof options === 'function') {
+        callback = options
+      } else if (options.cid && options.format && options.hashAlg) {
+        throw new Error('Can\'t put dag node. Please provide either `cid` OR `format` and `hashAlg` options.')
+      } else if ((options.format && !options.hashAlg) || (!options.format && options.hashAlg)) {
+        throw new Error('Can\'t put dag node. Please provide `format` AND `hashAlg` options.')
```

Comment:
Sorry, yes I completely missed that - you must `return callback(new Error(...))` for these